### PR TITLE
optimizate auto upgrade golang logic

### PR DESCRIPTION
**Pull request by ty-dc**
#### What type of PR is this?

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
close #4052

### CODE REVIEW

File: .github/workflows/update-golang-version.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
       - name: Compare versions and update Makefile.version
         run: |
           LATEST_GOLANG_VERSION=$(echo $LATEST_GOLANG_VERSION | sed 's/go//')
```
Comment (weizhoublue) [inline .github/workflows/update-golang-version.yaml:35]: 尤其是 shell 代码，一定要健壮，不能想当然
例如 30 行  “ CURRENT_GO_VERSION=$(grep '^GO_VERSION := ' Makefile.version | awk '{print $3}') ”

如果下次 谁修改了 “ GO_VERSION :=”， 加了个空格，少了个空格， CURRENT_GO_VERSION 为空，没有报错么 ？

Comment (ty-dc) [reply .github/workflows/update-golang-version.yaml]: done

File: .github/workflows/update-golang-version.yaml [hunk 2]
```diff
@@ -21,21 +21,34 @@ jobs:
...
+            echo "Error: Unable to get version number from LATEST_GOLANG_VERSION: ${LATEST_GOLANG_VERSION}"
+            exit 1
+          fi
           echo "LATEST_GOLANG_VERSION=${LATEST_GOLANG_VERSION}" >> $GITHUB_ENV
 
       - name: Read current Go version from Makefile.version
         id: read-makefile-version
         run: |
-          CURRENT_GO_VERSION=$(grep '^GO_VERSION := ' Makefile.version | awk '{print $3}')
+          CURRENT_GO_VERSION=$(grep '^GO_VERSION' Makefile.version | awk '{print $3}')
```
Comment (weizhoublue) [inline .github/workflows/update-golang-version.yaml:35]: 一行就提取出来了 

CURRENT_GO_VERSION=$( grep -E "^[[:space:]]*GO_VERSION[[:space:]]*:=" Makefile.version | awk -F '=' '{print $2}' | tr -d ' ' )

Comment (ty-dc) [reply .github/workflows/update-golang-version.yaml]: done

**Review by weizhoublue — APPROVED** [review]

**Pull request merged**